core/client/runtime/pom.xml

```diff
@@ -34,6 +34,14 @@
   </properties>
 
   <dependencies>
+    <!-- External dependencies -->
+    <!-- Have hadoop-client dependency in provided scope by default -->
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-client</artifactId>
+      <scope>provided</scope>
+    </dependency>
```

Comment:
transitive dependencies from this dep will be marked as provided too, and thus excluded from uber jar